core/src/main/scala/kafka/server/ConfigHelper.scala

```diff
@@ -159,7 +160,9 @@ class ConfigHelper(metadataCache: MetadataCache, config: KafkaConfig, configRepo
               throw new InvalidRequestException("Group name must not be empty")
             } else {
               val groupProps = configRepository.groupConfig(group)
-              val groupConfig = GroupConfig.fromProps(config.groupCoordinatorConfig.extractGroupConfigMap, groupProps)
+              val defaultConfigs: util.Map[String, Integer] = config.groupCoordinatorConfig.extractConsumerGroupConfigMap()
+              defaultConfigs.putAll(config.shareGroupConfig.extractShareGroupConfigMap())
+              val groupConfig = GroupConfig.fromProps(defaultConfigs, groupProps)
```

Comment:
Just thinking: wouldn't it be better to keep `extractGroupConfigMap` method which internall calls `extractConsumerGroupConfigMap` and `extractShareGroupConfigMap`? You won't be required above changes then. And look more clean. wdyt?

Reply:
Thanks for the review. I agree with you, but I wasn't really sure about the positioning of the extractGroupConfigMap method, that's why I left it for the next PR where I have a dedicated position for it. If it's possible, can we wait for the next PR to have this resolved ? @apoorvmittal10

Reply:
Ok, will wait.